# Table of Contents
 <p><div class="lev1"><a href="#PREDICTING-LOAN-REPAYMENT"><span class="toc-item-num">1&nbsp;&nbsp;</span>PREDICTING LOAN REPAYMENT</a></div><div class="lev2"><a href="#Problem-1.1---Preparing-the-Dataset"><span class="toc-item-num">1.1&nbsp;&nbsp;</span>Problem 1.1 - Preparing the Dataset</a></div><div class="lev2"><a href="#Problem-1.3---Preparing-the-Dataset"><span class="toc-item-num">1.2&nbsp;&nbsp;</span>Problem 1.3 - Preparing the Dataset</a></div><div class="lev3"><a href="#Explanation"><span class="toc-item-num">1.2.1&nbsp;&nbsp;</span>Explanation</a></div><div class="lev2"><a href="#Problem-1.4---Preparing-the-Dataset"><span class="toc-item-num">1.3&nbsp;&nbsp;</span>Problem 1.4 - Preparing the Dataset</a></div><div class="lev3"><a href="#Explanation"><span class="toc-item-num">1.3.1&nbsp;&nbsp;</span>Explanation</a></div><div class="lev2"><a href="#Problem-2.1---Prediction-Models"><span class="toc-item-num">1.4&nbsp;&nbsp;</span>Problem 2.1 - Prediction Models</a></div><div class="lev2"><a href="#Problem-2.2---Prediction-Models"><span class="toc-item-num">1.5&nbsp;&nbsp;</span>Problem 2.2 - Prediction Models</a></div><div class="lev3"><a href="#Explanation-1"><span class="toc-item-num">1.5.1&nbsp;&nbsp;</span>Explanation 1</a></div><div class="lev3"><a href="#Explanation"><span class="toc-item-num">1.5.2&nbsp;&nbsp;</span>Explanation</a></div><div class="lev2"><a href="#Problem-2.3---Prediction-Models"><span class="toc-item-num">1.6&nbsp;&nbsp;</span>Problem 2.3 - Prediction Models</a></div><div class="lev2"><a href="#Problem-2.4---Prediction-Models"><span class="toc-item-num">1.7&nbsp;&nbsp;</span>Problem 2.4 - Prediction Models</a></div><div class="lev2"><a href="#Problem-3.1---A-&quot;Smart-Baseline&quot;"><span class="toc-item-num">1.8&nbsp;&nbsp;</span>Problem 3.1 - A "Smart Baseline"</a></div><div class="lev2"><a href="#Problem-3.2---A-&quot;Smart-Baseline&quot;"><span class="toc-item-num">1.9&nbsp;&nbsp;</span>Problem 3.2 - A "Smart Baseline"</a></div><div class="lev2"><a href="#Problem-3.3---A-&quot;Smart-Baseline&quot;"><span class="toc-item-num">1.10&nbsp;&nbsp;</span>Problem 3.3 - A "Smart Baseline"</a></div><div class="lev2"><a href="#Problem-4.1---Computing-the-Profitability-of-an-Investment"><span class="toc-item-num">1.11&nbsp;&nbsp;</span>Problem 4.1 - Computing the Profitability of an Investment</a></div><div class="lev2"><a href="#Problem-4.2---Computing-the-Profitability-of-an-Investment"><span class="toc-item-num">1.12&nbsp;&nbsp;</span>Problem 4.2 - Computing the Profitability of an Investment</a></div><div class="lev2"><a href="#Problem-4.3---Computing-the-Profitability-of-an-Investment"><span class="toc-item-num">1.13&nbsp;&nbsp;</span>Problem 4.3 - Computing the Profitability of an Investment</a></div><div class="lev2"><a href="#Problem-5.1---A-Simple-Investment-Strategy"><span class="toc-item-num">1.14&nbsp;&nbsp;</span>Problem 5.1 - A Simple Investment Strategy</a></div>

PREDICTING LOAN REPAYMENT
==========================

In the lending industry, investors provide loans to borrowers in exchange for the promise of repayment with interest. If the borrower repays the loan, then the lender profits from the interest. However, if the borrower is unable to repay the loan, then the lender loses money. Therefore, lenders face the problem of predicting the risk of a borrower being unable to repay a loan.

To address this problem, we will use publicly available data from LendingClub.com, a website that connects borrowers and investors over the Internet. This dataset represents 9,578 3-year loans that were funded through the LendingClub.com platform between May 2007 and February 2010. The binary dependent variable not_fully_paid indicates that the loan was not paid back in full (the borrower either defaulted or the loan was "charged off," meaning the borrower was deemed unlikely to ever pay it back).

To predict this dependent variable, we will use the following independent variables available to the investor when deciding whether to fund a loan:

- credit.policy: 1 if the customer meets the credit underwriting criteria of LendingClub.com, and 0 otherwise.
- purpose: The purpose of the loan (takes values "credit_card", "debt_consolidation", "educational", "major_purchase", "small_business", and "all_other").
- int.rate: The interest rate of the loan, as a proportion (a rate of 11% would be stored as 0.11). Borrowers judged by LendingClub.com to be more risky are assigned higher interest rates.
- installment: The monthly installments ($) owed by the borrower if the loan is funded.
- log.annual.inc: The natural log of the self-reported annual income of the borrower.
- dti: The debt-to-income ratio of the borrower (amount of debt divided by annual income).
- fico: The FICO credit score of the borrower.
- days.with.cr.line: The number of days the borrower has had a credit line.
- revol.bal: The borrower's revolving balance (amount unpaid at the end of the credit card billing cycle).
- revol.util: The borrower's revolving line utilization rate (the amount of the credit line used relative to total credit available).
- inq.last.6mths: The borrower's number of inquiries by creditors in the last 6 months.
- delinq.2yrs: The number of times the borrower had been 30+ days past due on a payment in the past 2 years.
- pub.rec: The borrower's number of derogatory public records (bankruptcy filings, tax liens, or judgments).

## Problem 1.1 - Preparing the Dataset

Load the dataset loans.csv into a data frame called loans, and explore it using the str() and summary() functions.

What proportion of the loans in the dataset were not paid in full? Please input a number between 0 and 1.

In [1]:
loans<- read.csv("loans.csv")
str(loans)
summary(loans)

'data.frame':	9578 obs. of  14 variables:
 $ credit.policy    : int  1 1 1 1 1 1 1 1 1 1 ...
 $ purpose          : Factor w/ 7 levels "all_other","credit_card",..: 3 2 3 3 2 2 3 1 5 3 ...
 $ int.rate         : num  0.119 0.107 0.136 0.101 0.143 ...
 $ installment      : num  829 228 367 162 103 ...
 $ log.annual.inc   : num  11.4 11.1 10.4 11.4 11.3 ...
 $ dti              : num  19.5 14.3 11.6 8.1 15 ...
 $ fico             : int  737 707 682 712 667 727 667 722 682 707 ...
 $ days.with.cr.line: num  5640 2760 4710 2700 4066 ...
 $ revol.bal        : int  28854 33623 3511 33667 4740 50807 3839 24220 69909 5630 ...
 $ revol.util       : num  52.1 76.7 25.6 73.2 39.5 51 76.8 68.6 51.1 23 ...
 $ inq.last.6mths   : int  0 0 1 1 0 0 0 0 1 1 ...
 $ delinq.2yrs      : int  0 0 0 0 1 0 0 0 0 0 ...
 $ pub.rec          : int  0 0 0 0 0 0 1 0 0 0 ...
 $ not.fully.paid   : int  0 0 0 0 0 0 1 1 0 0 ...


 credit.policy                 purpose        int.rate       installment    
 Min.   :0.000   all_other         :2331   Min.   :0.0600   Min.   : 15.67  
 1st Qu.:1.000   credit_card       :1262   1st Qu.:0.1039   1st Qu.:163.77  
 Median :1.000   debt_consolidation:3957   Median :0.1221   Median :268.95  
 Mean   :0.805   educational       : 343   Mean   :0.1226   Mean   :319.09  
 3rd Qu.:1.000   home_improvement  : 629   3rd Qu.:0.1407   3rd Qu.:432.76  
 Max.   :1.000   major_purchase    : 437   Max.   :0.2164   Max.   :940.14  
                 small_business    : 619                                    
 log.annual.inc        dti              fico       days.with.cr.line
 Min.   : 7.548   Min.   : 0.000   Min.   :612.0   Min.   :  179    
 1st Qu.:10.558   1st Qu.: 7.213   1st Qu.:682.0   1st Qu.: 2820    
 Median :10.928   Median :12.665   Median :707.0   Median : 4140    
 Mean   :10.932   Mean   :12.607   Mean   :710.8   Mean   : 4562    
 3rd Qu.:11.290   3rd Qu.:17.950   3rd 

In [2]:
table(loans$not.fully.paid)


   0    1 
8045 1533 

In [6]:
table(loans$not.fully.paid)/nrow(loans)


        0         1 
0.8399457 0.1600543 

The following variables present missing values: log.annual.inc , days.with.cr.line , revol.util, inq.last.6mths, delinq.2yrs, pub.rec .

## Problem 1.3 - Preparing the Dataset

Which of the following is the best reason to fill in the missing values for these variables instead of removing observations with missing data? 

(Hint: you can use the subset() function to build a data frame with the observations missing at least one value. To test if a variable, for example pub.rec, is missing a value, use is.na(pub.rec).)

- If we remove the missing observations there will be too little remaining data, leading to overfitting in our models.
- We want to be able to predict risk for all borrowers, instead of just the ones with all data reported.
- In this dataset the observations with missing data have a much different rate of not paying in full, so removing them would bias subsequent models.

### Explanation

Answering this question requires analyzing the loans with missing data. 

We can build a data frame limited to observations with some missing data with the following command:

> missing = subset(loans, is.na(log.annual.inc) | is.na(days.with.cr.line) | is.na(revol.util) | is.na(inq.last.6mths) | is.na(delinq.2yrs) | is.na(pub.rec))

From nrow(missing), we see that only 62 of 9578 loans have missing data; removing this small number of observations would not lead to overfitting. 

From table(missing$not.fully.paid), we see that 12 of 62 loans with missing data were not fully paid, or 19.35%. 
This rate is similar to the 16.01% across all loans, so the form of biasing described is not an issue. However, to predict risk for loans with missing data we need to fill in the missing values instead of removing the observations.

## Problem 1.4 - Preparing the Dataset

For the rest of this problem, we'll be using a revised version of the dataset that has the missing values filled in with multiple imputation (which was discussed in the Recitation of this Unit). To ensure everybody has the same data frame going forward, you can either run the commands below in your R console (if you haven't already, run the command install.packages("mice") first), or you can download and load into R the dataset we created after running the imputation: loans_imputed.csv.

IMPORTANT NOTE: On certain operating systems, the imputation results are not the same even if you set the random seed. If you decide to do the imputation yourself, please still read the provided imputed dataset (loans_imputed.csv) into R and compare your results, using the summary function. If the results are different, please make sure to use the data in loans_imputed.csv for the rest of the problem.

library(mice)

set.seed(144)

vars.for.imputation = setdiff(names(loans), "not.fully.paid")

imputed = complete(mice(loans[vars.for.imputation]))

loans[vars.for.imputation] = imputed

Note that to do this imputation, we set vars.for.imputation to all variables in the data frame except for not.fully.paid, to impute the values using all of the other independent variables.

What best describes the process we just used to handle missing values?


- We removed all observations with a missing value.
- We filled each missing value with the average of all other values for that variable.
- We predicted missing variable values using the available independent variables for each observation.
- We predicted missing variable values using the available independent and dependent variables for each observation.

In [7]:
library(mice)
set.seed(144)
vars.for.imputation = setdiff(names(loans), "not.fully.paid")
imputed = complete(mice(loans[vars.for.imputation]))
loans[vars.for.imputation] = imputed

Loading required package: Rcpp
mice 2.25 2015-11-09



 iter imp variable
  1   1  log.annual.inc  days.with.cr.line  revol.util  inq.last.6mths  delinq.2yrs  pub.rec
  1   2  log.annual.inc  days.with.cr.line  revol.util  inq.last.6mths  delinq.2yrs  pub.rec
  1   3  log.annual.inc  days.with.cr.line  revol.util  inq.last.6mths  delinq.2yrs  pub.rec
  1   4  log.annual.inc  days.with.cr.line  revol.util  inq.last.6mths  delinq.2yrs  pub.rec
  1   5  log.annual.inc  days.with.cr.line  revol.util  inq.last.6mths  delinq.2yrs  pub.rec
  2   1  log.annual.inc  days.with.cr.line  revol.util  inq.last.6mths  delinq.2yrs  pub.rec
  2   2  log.annual.inc  days.with.cr.line  revol.util  inq.last.6mths  delinq.2yrs  pub.rec
  2   3  log.annual.inc  days.with.cr.line  revol.util  inq.last.6mths  delinq.2yrs  pub.rec
  2   4  log.annual.inc  days.with.cr.line  revol.util  inq.last.6mths  delinq.2yrs  pub.rec
  2   5  log.annual.inc  days.with.cr.line  revol.util  inq.last.6mths  delinq.2yrs  pub.rec
  3   1  log.annual.inc  days.with.cr.line  revol.

### Explanation

Imputation predicts missing variable values for a given observation using the variable values that are reported. We called the imputation on a data frame with the dependent variable not.fully.paid removed, so we predicted the missing values using only other independent variables.

So, we predicted missing variable values using the available independent variables for each observation.

## Problem 2.1 - Prediction Models

Now that we have prepared the dataset, we need to split it into a training and testing set. To ensure everybody obtains the same split, set the random seed to 144 (even though you already did so earlier in the problem) and use the sample.split function to select the 70% of observations for the training set (the dependent variable for sample.split is not.fully.paid). Name the data frames train and test.

Now, use logistic regression trained on the training set to predict the dependent variable not.fully.paid using all the independent variables.

Which independent variables are significant in our model? (Significant variables have at least one star, or a Pr(>|z|) value less than 0.05.) Select all that apply.

In [8]:
set.seed(144)
library(caTools)

In [9]:
?sample.split

In [10]:
split <- sample.split(loans$not.fully.paid, SplitRatio=.7)

In [11]:
train <- subset(loans, split == TRUE)
test <- subset(loans, split == FALSE)

In [12]:
nrow(train)/nrow(loans)
nrow(test)/nrow(loans)

[1] 0.7000418

[1] 0.2999582

In [13]:
?glm

In [14]:
mod <- glm(not.fully.paid ~ ., data=train, family=binomial)
summary(mod)


Call:
glm(formula = not.fully.paid ~ ., family = binomial, data = train)

Deviance Residuals: 
    Min       1Q   Median       3Q      Max  
-2.2097  -0.6214  -0.4950  -0.3601   2.6414  

Coefficients:
                    Estimate Std. Error z value Pr(>|z|)    
(Intercept)        9.260e+00  1.554e+00   5.958 2.55e-09 ***
credit.policy     -3.327e-01  1.011e-01  -3.292 0.000995 ***
purpose2          -6.100e-01  1.344e-01  -4.538 5.67e-06 ***
purpose3          -3.181e-01  9.185e-02  -3.463 0.000534 ***
purpose4           1.386e-01  1.753e-01   0.791 0.429074    
purpose5           1.774e-01  1.479e-01   1.199 0.230496    
purpose6          -4.783e-01  2.009e-01  -2.381 0.017260 *  
purpose7           4.159e-01  1.419e-01   2.932 0.003373 ** 
int.rate           6.202e-01  2.085e+00   0.297 0.766111    
installment        1.279e-03  2.093e-04   6.110 9.96e-10 ***
log.annual.inc    -4.357e-01  7.151e-02  -6.093 1.11e-09 ***
dti                4.733e-03  5.501e-03   0.861 0.389508    
fico

In [21]:
subset(summary(mod)$coefficients[,"Pr(>|z|)"], summary(mod)$coefficients[,"Pr(>|z|)"] < .05)

(Intercept)  credit.policy       purpose2       purpose3       purpose6 
  2.550660e-09   9.948617e-04   5.666057e-06   5.342783e-04   1.726005e-02 
      purpose7    installment log.annual.inc           fico      revol.bal 
  3.372613e-03   9.959934e-10   1.110724e-09   3.598345e-08   7.966422e-03 
inq.last.6mths        pub.rec 
  1.055407e-07   4.898756e-03

## Problem 2.2 - Prediction Models

Consider two loan applications, which are identical other than the fact that the borrower in Application A has FICO credit score 700 while the borrower in Application B has FICO credit score 710.

Let Logit(A) be the log odds of loan A not being paid back in full, according to our logistic regression model, and define Logit(B) similarly for loan B. What is the value of Logit(A) - Logit(B)?

In [27]:
summary(mod)$coefficients["fico","Estimate"] * -10

[1] 0.09405628

### Explanation 1
Because Application A is identical to Application B other than having a FICO score 10 lower, its predicted log odds differ by $-0.009317 * -10 = 0.09317$ from the predicted log odds of Application B.


--------------

Now, let O(A) be the odds of loan A not being paid back in full, according to our logistic regression model, and define O(B) similarly for loan B. What is the value of O(A)/O(B)? (HINT: Use the mathematical rule that exp(A + B + C) = exp(A)*exp(B)*exp(C). Also, remember that exp() is the exponential function in R.)

In [28]:
exp(summary(mod)$coefficients["fico","Estimate"] * -10)

[1] 1.098622

### Explanation

Using the answer from the previous question, the predicted odds of loan A not being paid back in full are exp(0.09317) = 1.0976 times larger than the predicted odds for loan B. Intuitively, it makes sense that loan A should have higher odds of non-payment than loan B, since the borrower has a worse credit score.

## Problem 2.3 - Prediction Models

Predict the probability of the test set loans not being paid back in full (remember type="response" for the predict function). Store these predicted probabilities in a variable named predicted.risk and add it to your test set (we will use this variable in later parts of the problem). Compute the confusion matrix using a threshold of 0.5.

What is the accuracy of the logistic regression model? Input the accuracy as a number between 0 and 1.

In [29]:
?predict

In [31]:
test$predicted.risk <- predict(mod, newdata=test, type="response")

Warning message:
"contrasts dropped from factor purpose"

In [32]:
thresh <- .5
table(test$not.fully.paid, test$predicted.risk > thresh)

   
    FALSE TRUE
  0  2400   13
  1   457    3

In [34]:
accuracy <- (2400 + 3) / (2400 + 13 + 457 +3)
accuracy

[1] 0.8364079

What is the accuracy of the baseline model? Input the accuracy as a number between 0 and 1.

In [35]:
table(test$not.fully.paid)


   0    1 
2413  460 

In [37]:
baseline_accuracy <- 2413/(2413+460)
baseline_accuracy

[1] 0.8398886

## Problem 2.4 - Prediction Models

Use the ROCR package to compute the test set AUC.

In [38]:
library(ROCR)

Loading required package: gplots

Attaching package: 'gplots'

The following object is masked from 'package:stats':

    lowess



In [39]:
pred = prediction(test$predicted.risk, test$not.fully.paid)
as.numeric(performance(pred, "auc")@y.values)

[1] 0.6721337

## Problem 3.1 - A "Smart Baseline"

In the previous problem, we built a logistic regression model that has an AUC significantly higher than the AUC of 0.5 that would be obtained by randomly ordering observations.

However, LendingClub.com assigns the interest rate to a loan based on their estimate of that loan's risk. This variable, int.rate, is an independent variable in our dataset. In this part, we will investigate using the loan's interest rate as a "smart baseline" to order the loans according to risk.

Using the training set, build a bivariate logistic regression model (aka a logistic regression model with a single independent variable) that predicts the dependent variable not.fully.paid using only the variable int.rate.

The variable int.rate is highly significant in the bivariate model, but it is not significant at the 0.05 level in the model trained with all the independent variables. What is the most likely explanation for this difference?

- int.rate is correlated with other risk-related variables, and therefore does not incrementally improve the model when those other variables are included.
- This effect is likely due to the training/testing set split we used. In other splits, we could see the opposite effect.
- These models are trained on a different set of observations, so the coefficients are not comparable.

In [40]:
mod_biv <- glm(not.fully.paid ~ int.rate, data=train, family=binomial)

In [41]:
summary(mod_biv)


Call:
glm(formula = not.fully.paid ~ int.rate, family = binomial, data = train)

Deviance Residuals: 
    Min       1Q   Median       3Q      Max  
-1.0547  -0.6271  -0.5442  -0.4361   2.2914  

Coefficients:
            Estimate Std. Error z value Pr(>|z|)    
(Intercept)  -3.6726     0.1688  -21.76   <2e-16 ***
int.rate     15.9214     1.2702   12.54   <2e-16 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

(Dispersion parameter for binomial family taken to be 1)

    Null deviance: 5896.6  on 6704  degrees of freedom
Residual deviance: 5734.8  on 6703  degrees of freedom
AIC: 5738.8

Number of Fisher Scoring iterations: 4


Decreased significance between a bivariate and multivariate model is typically due to correlation.

From cor(train\$int.rate, train\$fico), we can see that the interest rate is moderately well correlated with a borrower's credit score.

Training/testing set split rarely has a large effect on the significance of variables (this can be verified in this case by trying out a few other training/testing splits), and the models were trained on the same observations.

In [42]:
cor(train$int.rate, train$fico)

[1] -0.711659

## Problem 3.2 - A "Smart Baseline"

Make test set predictions for the bivariate model. What is the highest predicted probability of a loan not being paid in full on the testing set?

In [43]:
pred_biv <- predict(mod_biv, newdata=test, type="response")
summary(pred_biv)

   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
0.06196 0.11550 0.15080 0.15960 0.18930 0.42660 

With a logistic regression cutoff of 0.5, how many loans would be predicted as not being paid in full on the testing set?

$\Rightarrow$ According to the summary function, the maximum predicted probability of the loan not being paid back is 0.4266, which means no loans would be flagged at a logistic regression cutoff of 0.5.

## Problem 3.3 - A "Smart Baseline"

What is the test set AUC of the bivariate model?

In [45]:
pred = prediction(pred_biv, test$not.fully.paid)
as.numeric(performance(pred, "auc")@y.values)

[1] 0.6239081

## Problem 4.1 - Computing the Profitability of an Investment

While thus far we have predicted if a loan will be paid back or not, an investor needs to identify loans that are expected to be profitable.

If the loan is paid back in full, then the investor makes interest on the loan. However, if the loan is not paid back, the investor loses the money invested. Therefore, the investor should seek loans that best balance this risk and reward.

To compute interest revenue, consider a \$c investment in a loan that has an annual interest rate r over a period of t years. Using continuous compounding of interest, this investment pays back $c * exp(rt)$ dollars by the end of the t years, where exp(rt) is e raised to the r*t power.

How much does a \$10 investment with an annual interest rate of 6% pay back after 3 years, using continuous compounding of interest? Hint: remember to convert the percentage to a proportion before doing the math. Enter the number of dollars, without the $ sign.

In [53]:
10*exp(0.06*3)

[1] 11.97217

## Problem 4.2 - Computing the Profitability of an Investment

While the investment has value c * exp(rt) dollars after collecting interest, the investor had to pay $c for the investment. What is the profit to the investor if the investment is paid back in full?

- $c * exp(rt) - c$
- $c * exp(rt)$
- $c * exp(rt) + c$
- $-c$
- $0$
- $c$

A person's profit is what they get minus what they paid for it. In this case, the investor gets c * exp(rt) but paid c, yielding a profit of $c * exp(rt) - c$.

## Problem 4.3 - Computing the Profitability of an Investment

Now, consider the case where the investor made a $c investment, but it was not paid back in full. Assume, conservatively, that no money was received from the borrower (often a lender will receive some but not all of the value of the loan, making this a pessimistic assumption of how much is received). What is the profit to the investor in this scenario?

- $c * exp(rt) - c$
- $c * exp(rt)$
- $c * exp(rt) + c$
- $-c$
- $0$
- $c$

Explanation
A person's profit is what they get minus what they paid for it. In this case, the investor gets no money but paid c dollars, yielding a profit of $-c$ dollars.

## Problem 5.1 - A Simple Investment Strategy

In the previous subproblem, we concluded that an investor who invested c dollars in a loan with interest rate r for t years makes c * (exp(rt) - 1) dollars of profit if the loan is paid back in full and -c dollars of profit if the loan is not paid back in full (pessimistically).

In order to evaluate the quality of an investment strategy, we need to compute this profit for each loan in the test set. For this variable, we will assume a $1 investment (aka c=1). To create the variable, we first assign to the profit for a fully paid loan, exp(rt)-1, to every observation, and we then replace this value with -1 in the cases where the loan was not paid in full. All the loans in our dataset are 3-year loans, meaning t=3 in our calculations. Enter the following commands in your R console to create this new variable:

test\$profit = exp(test$int.rate*3) - 1

test\$profit[test$not.fully.paid == 1] = -1

What is the maximum profit of a $10 investment in any loan in the testing set (do not include the $ sign in your answer)?